File: hw5/examples/python/mysql_test_2.py

import mysql.connector as mc

import config

def main():
    try:
        # get connection info
        hst = config.host
        dat = config.database
        usr = config.user
        pwd = config.password
        
        # create connection
        cn = mc.connect(host=hst, database=dat, user=usr, password=pwd)
        
        # get a category from the user
        user_input_1 = input('Please enter a pet type (dog, cat, etc): ')
        user_input_2 = input('Please enter an appearance keyword: ') 

        # result set
        rs = cn.cursor()
        q = ('SELECT id, name, appearance '
             'FROM pet '
             'WHERE type = %s AND INSTR(appearance, %s)')

        # execute the query
        rs.execute(q, (user_input_1, user_input_2))
        
        # display results
        for row in rs:
            print(f'{row[0]}, {row[1]}, {row[2]}') 
            
    except mc.Error as err:
        print(err)

    finally:
        rs.close()
        cn.close()
            
        
if __name__ == '__main__':
    main()

File: hw5/examples/cpp/mysql_test_2.cpp

#include <iostream>
#include <mysql_connection.h>
#include <cppconn/driver.h>
#include <cppconn/prepared_statement.h>
#include <cppconn/resultset.h>
#include <cppconn/exception.h>

#include "config.h"

using namespace std;


int main()
{
  try {

    // connection info
    string hst = config::host;
    string usr = config::user;
    string pwd = config::password;
    string dab = config::database;

    // create connection
    sql::Driver* driver = get_driver_instance();
    sql::Connection* cn = driver->connect(hst, usr, pwd);
    cn->setSchema(dab);

    // get a category from the user
    string user_input_1;
    string user_input_2;
    cout << "Please enter a pet type (dog, cat, etc): ";
    cin >> user_input_1;
    cout << "Please enter an appearance keyword: ";
    cin >> user_input_2;
    
    // create statement
    string q =
      "SELECT id, name, appearance "
      "FROM pet "
      "WHERE type = ? AND INSTR(appearance, ?)";
    sql::PreparedStatement* st = cn->prepareStatement(q);
    st->setString(1, user_input_1);
    st->setString(2, user_input_2);
    
    // execute query
    sql::ResultSet* rs = st->executeQuery();

    // print result
    while(rs->next()) {
      cout << rs->getInt("id") << ", "
           << rs->getString("name") << ", "
           << rs->getString("appearance") << endl;
    }

    // clean up
    delete rs;
    delete st;
    delete cn;

  } catch(sql::SQLException& e) {
    cout << e.what() << endl;
  }
  
}
